proposals/CHAOSS-Improving-Open-Source-Sustainability.md  

```diff
@@ -0,0 +1,124 @@
+
+Discovery: The process of identifying related open source software components is ad-hoc, leading to an unclear picture of software present within an ecosystem.
+Health Assessment: Software can be available without a healthy, active contributor community to keep it useful and synchronized with software around it. Without sustained effort software can create problems in the supply chain of nearby open-source software projects. These characteristics include Identifying ecosystem-level characteristics that correspond with reliability, safety, security, inclusivity, and what might generally be described as health trajectories. 
+
+As users coalesce around open-source software toolchains, support and skill flow also coalesce, generating valuable, and stabilizing network effects. If developers and maintainers of open-source software know who is using the software and how it is being used and combined in workflows and infrastructure, then they will be better able to support their users. In turn, this will help ensure compatibility with nearby software, and anticipate the future software needs of cutting-edge projects. 
+
+More subtle effects are also valuable. For example, clearer understanding of dependencies and patterns of use in software make local ecosystems more able to compose software from smaller packages, each more specialized. Smaller, more specialized packages increase “option value” or the innovative potential from recombination. An increased option value can help fix problems or provide new capabilities in fewer locations, meaning fewer partial, redundant, and desynchronized changes.
+
+The first part in this project will scale the dependency analysis and machine learning components of the CHAOSS/Augur software project. Specifically, we will engage the CHAOSS community in an iterative process of identifying critical sub-ecosystems in package managed libraries, core corporatized infrastructure, and scientific software. We will then scale CHAOSS/Augur’s existing machine learning algorithms to discover, assess, and identify ecosystem-level characteristics that generate an ecosystem map. The map will substantially accelerate the open-source community writ large’s ability to discover opportunities for improvement, and models of practice that are effective in similar ecosystems and sub-ecosystems. 
```

Comment:
What kind of dependency analysis does Augur already do? What data sources does it get this information from? Do you plan to integrate other dependency analysis tools / APIs or just use Augur?

Reply:
Augur does three types of dependency analysis: 
1. We check every packaged managed library against the most recent stable release, then we calculate "libyears" for each library". This is a CHAOSS metric that provides a useful approximation of how current the package managed libraries are in a project. 
2. We enumerate all dependencies for the 10 common languages, whether or not they are package managed. 
3. We apply the OSSF Scorecard to each repository. This particular design stores the data in a single JSON field to accommodate the ongoing development of the scorecard without having it break.